packages/video_player/video_player_windows/windows/test/mocks.h

```diff
@@ -0,0 +1,107 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef PACKAGES_VIDEO_PLAYER_VIDEO_PLAYER_WINDOWS_WINDOWS_TEST_MOCKS_H_
+#define PACKAGES_VIDEO_PLAYER_VIDEO_PLAYER_WINDOWS_WINDOWS_TEST_MOCKS_H_
+
+#include <flutter/method_call.h>
+#include <flutter/method_result_functions.h>
+#include <flutter/standard_method_codec.h>
+#include <flutter/texture_registrar.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <mfcaptureengine.h>
+
+#include "video_player.h"
+
+namespace video_player_windows {
+namespace test {
+
+namespace {
+
+using flutter::EncodableMap;
+using flutter::EncodableValue;
+using ::testing::_;
+
+class MockBinaryMessenger : public flutter::BinaryMessenger {
+ public:
+  ~MockBinaryMessenger() = default;
+
+  MOCK_METHOD(void, Send,
+              (const std::string& channel, const uint8_t* message,
+               size_t message_size, flutter::BinaryReply reply),
+              (const));
+
+  MOCK_METHOD(void, SetMessageHandler,
+              (const std::string& channel,
+               flutter::BinaryMessageHandler handler),
+              ());
+};
+
+class MockTextureRegistrar : public flutter::TextureRegistrar {
+ public:
+  MockTextureRegistrar() {
+    ON_CALL(*this, RegisterTexture)
+        .WillByDefault([this](flutter::TextureVariant* texture) -> int64_t {
+          EXPECT_TRUE(texture);
+          this->texture_ = texture;
+          this->texture_id_ = 1000;
+          return this->texture_id_;
+        });
+
+    // Deprecated pre-Flutter-3.4 version.
```

Comment:
We aren't supporting any versions of Flutter before 3.4 in this plugin, so there's cruft here that can be removed.

Reply:
Done.